irispy/new_sji.py

```diff
@@ -0,0 +1,179 @@
+        File name to be read
+
+    Returns
+    -------
+    result: 'irispy.sji.SJICube'
+
+    """
+
+    # Open a fits file
+    my_file = fits.open(filename, memmap=True, do_not_scale_image_data=True)
+    # Derive WCS, data and mask for NDCube from fits file.
+    wcs = WCS(my_file[0].header)
+    data = my_file[0].data
+    data_nan_masked = my_file[0].data
+    data_nan_masked[data == BAD_PIXEL_VALUE_UNSCALED] = BAD_PIXEL_VALUE
+    mask = data_nan_masked == BAD_PIXEL_VALUE_UNSCALED
+    # Derive unit and readout noise from detector.
+    exposure_times = my_file[1].data[:, my_file[1].header["EXPTIMES"]]
+    unit = iris_tools.DN_UNIT["SJI"]
+    readout_noise = iris_tools.READOUT_NOISE["SJI"]
+    # Derive uncertainty of data for NDCube from fits file.
+    uncertainty = u.Quantity(np.sqrt((data_nan_masked*unit).to(u.photon).value
+                                     + readout_noise.to(u.photon).value**2),
+                             unit=u.photon).to(unit).value
+    # Derive extra coordinates for NDCube from fits file.
+    times = np.array([parse_time(my_file[0].header["STARTOBS"])
+                      + timedelta(seconds=s)
+                      for s in my_file[1].data[:, my_file[1].header["TIME"]]])
+    pztx = my_file[1].data[:, my_file[1].header["PZTX"]] * u.arcsec
+    pzty = my_file[1].data[:, my_file[1].header["PZTY"]] * u.arcsec
+    xcenix = my_file[1].data[:, my_file[1].header["XCENIX"]] * u.arcsec
+    ycenix = my_file[1].data[:, my_file[1].header["YCENIX"]] * u.arcsec
+    obs_vrix = my_file[1].data[:, my_file[1].header["OBS_VRIX"]] * u.m/u.s
+    ophaseix = my_file[1].data[:, my_file[1].header["OPHASEIX"]]
+    extra_coords = [('TIME', 0, times), ("PZTX", 0, pztx), ("PZTY", 0, pzty),
+                    ("XCENIX", 0, xcenix), ("YCENIX", 0, ycenix),
+                    ("OBS_VRIX", 0, obs_vrix), ("OPHASEIX", 0, ophaseix),
+                    ("EXPOSURE TIME", 0, exposure_times)]
+    # Extraction of meta for NDCube from fits file.
+    date_obs = my_file[0].header.get('DATE_OBS', None)
+    date_obs = parse_time(date_obs) if date_obs else None
+    date_end = my_file[0].header.get('DATE_END', None)
+    date_end = parse_time(date_end) if date_end else None
+    startobs = my_file[0].header.get('STARTOBS', None)
+    startobs = parse_time(startobs) if startobs else None
+    endobs = my_file[0].header.get('ENDOBS', None)
+    endobs = parse_time(endobs) if endobs else None
+    meta = {'TELESCOP': my_file[0].header.get('TELESCOP', "Unknown"),
+            'INSTRUME': my_file[0].header.get('INSTRUME', "Unknown"),
+            'TWAVE1': my_file[0].header.get('TWAVE1', "Unknown"),
+            'DATE_OBS': date_obs,
+            'DATE_END': date_end,
+            'STARTOBS': startobs,
+            'ENDOBS': endobs,
+            'NBFRAMES': my_file[0].data.shape[0],
+            'OBSID': my_file[0].header.get('OBSID', "Unknown"),
+            'OBS_DESC': my_file[0].header.get('OBS_DESC', "Unknown")}
```

Comment:
Having thought about this, ```my_file[0].header.get('OBSID', "Unknown")``` should be ```my_file[0].header.get('OBSID', None)```.   Same goes for all other occurences of ```"Unknown"```.  That way you can use if statements like:
```
if not self.meta["X"]:
    # Do something
````
rather than explicitly having to check whether == "Unknown".

Reply:
Sure, this is better to use None if we'll create a test. I was using "Unknown" because this one is more relevant for a data.

Reply:
None is now replacing "Unknown"

Reply:
Looks like ```"Unknown"``` is still being used here.  Perhaps you have not pushed your changes?

Reply:
I don't have pushed now, I am still working on some problems that I try to solve. I will push after.